assets/js/src/wizard/woocommerce_import_controller.jsx

```diff
@@ -0,0 +1,81 @@
+import React from 'react';
+import MailPoet from 'mailpoet';
+import WelcomeWizardHeader from './header.jsx';
+import WooCommerceImportListStep from './steps/woo_commerce_import_list_step.jsx';
+
+class WooCommerceImportController extends React.Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      loading: false,
+    };
+    this.updateSettings = this.updateSettings.bind(this);
+    this.scheduleImport = this.scheduleImport.bind(this);
+    this.finishWizard = this.finishWizard.bind(this);
+    this.submit = this.submit.bind(this);
+  }
+
+  finishWizard() {
+    this.setState({ loading: true });
+    window.location = window.finish_wizard_url;
+  }
+
+  updateSettings(data) {
+    return MailPoet.Ajax.post({
+      api_version: window.mailpoet_api_version,
+      endpoint: 'settings',
+      action: 'set',
+      data,
+    }).fail((response) => {
```

Comment:
When you catch the error here it will resolve the promise and still try to run `scheduleImport` and `finishWizard` or am I wrong?

Reply:
I don't think so. If first promise in chain fails the ohers are not called. We use jQuery.Deffered so I rather double checked:

```javascript
(function () {
	let deffered = jQuery.Deferred();
	let prom = deffered.promise();

	prom.fail(function(e) { console.log('error', e);}).then(function() { console.log('success 1') });
	prom.then(function() { console.log('success 2'); });

	deffered.reject('error_value');
})();
```
This will never log any of the success logs.